vulnfeeds/cvss30/cvss30.go

```diff
@@ -0,0 +1,733 @@
+	ModifiedConfidentialityImpact *ModifiedCiaType `json:"modifiedConfidentialityImpact,omitempty" yaml:"modifiedConfidentialityImpact,omitempty" mapstructure:"modifiedConfidentialityImpact,omitempty"`
+
+	// ModifiedIntegrityImpact corresponds to the JSON schema field
+	// "modifiedIntegrityImpact".
+	ModifiedIntegrityImpact *ModifiedCiaType `json:"modifiedIntegrityImpact,omitempty" yaml:"modifiedIntegrityImpact,omitempty" mapstructure:"modifiedIntegrityImpact,omitempty"`
+
+	// ModifiedPrivilegesRequired corresponds to the JSON schema field
+	// "modifiedPrivilegesRequired".
+	ModifiedPrivilegesRequired *ModifiedPrivilegesRequiredType `json:"modifiedPrivilegesRequired,omitempty" yaml:"modifiedPrivilegesRequired,omitempty" mapstructure:"modifiedPrivilegesRequired,omitempty"`
+
+	// ModifiedScope corresponds to the JSON schema field "modifiedScope".
+	ModifiedScope *ModifiedScopeType `json:"modifiedScope,omitempty" yaml:"modifiedScope,omitempty" mapstructure:"modifiedScope,omitempty"`
+
+	// ModifiedUserInteraction corresponds to the JSON schema field
+	// "modifiedUserInteraction".
+	ModifiedUserInteraction *ModifiedUserInteractionType `json:"modifiedUserInteraction,omitempty" yaml:"modifiedUserInteraction,omitempty" mapstructure:"modifiedUserInteraction,omitempty"`
+
+	// PrivilegesRequired corresponds to the JSON schema field "privilegesRequired".
+	PrivilegesRequired *PrivilegesRequiredType `json:"privilegesRequired,omitempty" yaml:"privilegesRequired,omitempty" mapstructure:"privilegesRequired,omitempty"`
+
+	// RemediationLevel corresponds to the JSON schema field "remediationLevel".
+	RemediationLevel *RemediationLevelType `json:"remediationLevel,omitempty" yaml:"remediationLevel,omitempty" mapstructure:"remediationLevel,omitempty"`
+
+	// ReportConfidence corresponds to the JSON schema field "reportConfidence".
+	ReportConfidence *ConfidenceType `json:"reportConfidence,omitempty" yaml:"reportConfidence,omitempty" mapstructure:"reportConfidence,omitempty"`
+
+	// Scope corresponds to the JSON schema field "scope".
+	Scope *ScopeType `json:"scope,omitempty" yaml:"scope,omitempty" mapstructure:"scope,omitempty"`
+
+	// TemporalScore corresponds to the JSON schema field "temporalScore".
+	TemporalScore *ScoreType `json:"temporalScore,omitempty" yaml:"temporalScore,omitempty" mapstructure:"temporalScore,omitempty"`
+
+	// TemporalSeverity corresponds to the JSON schema field "temporalSeverity".
+	TemporalSeverity *SeverityType `json:"temporalSeverity,omitempty" yaml:"temporalSeverity,omitempty" mapstructure:"temporalSeverity,omitempty"`
+
+	// UserInteraction corresponds to the JSON schema field "userInteraction".
+	UserInteraction *UserInteractionType `json:"userInteraction,omitempty" yaml:"userInteraction,omitempty" mapstructure:"userInteraction,omitempty"`
+
+	// VectorString corresponds to the JSON schema field "vectorString".
+	VectorString string `json:"vectorString" yaml:"vectorString" mapstructure:"vectorString"`
```

Comment:
These structs are **incredibly** complicated. 

Do we actually need all of them? Or do we only need the `VectorString` here? If so, we can just replace all of this with a much simpler struct containing only this item, and omit everything else.

Reply:
Bear in mind:

1) this file is generated by go-jsonschema, and hasn't required any modifications (yet)
2) everything is appearing to work as-is

The struct is complicated because the schema is complicated. What harm is having a fully functional complicated struct? It's future-proof.

Reply:
It's harder to review and maintain, and adds a dependency on a tool which may or may not be maintained in the future when we need it again. 

I suggest we follow YAGNI here. If this all changes in the future it's possible everything will have to be re-done anyway (e.g. if CVE 5.0 is adopted by the NVD in their output).

Reply:
Removed in 779c090